include/range/v3/detail/config.hpp

```diff
@@ -294,6 +294,10 @@
 
 #endif  // RANGES_CXX_INLINE_VARIABLES
 
+#ifdef RANGES_DOXYGEN_INVOKED
+#define RANGES_INLINE_VARIABLE(type, name) \
+    inline constexpr type name{};
+#else
```

Comment:
I like the idea, but please move this out of the code and into the `PREDEFINED` variable in doc/Doxyfile.in. Ditto for the `CONCEPT_REQUIRES` stuff.